### PR TITLE
Feature/pagination

### PR DESCRIPTION
**Pull request by RubenDeAndrade**
I created a pagination component to navigate between list.
We can give it list length, page limit, and it displays the appropriate pagination.

When current page changed, a callback is fired to parent so we can display new elements in our list.

# Screenshot

Capture d’écran 2021-03-05 à 11 18 44

# Test

`yarn storybook`

### CODE REVIEW

**Comment by williamgoulois** [comment]
Code un peu complexe à lire et comprendre, j'ai pu tester sur storybook et je trouve que l'UX est bizarre pour une pagination (notamment quand on passe de la page 66 à 67 si on clique sur Next le bouton Next devient 67)

On changera l'UX plus tard

File: src/stories/pagination/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                    </li>
+                )
+                if (page === RIGHT_PAGE) return (
+                    <li key={index} className='page-link-wrapper' onClick={handleMoveRight}>
+                        Next
+                    </li>
+                )
+
+                const pageLinkClasses = cx('page-link-wrapper', {
+                    ['page-link-wrapper--active']: currentPage === page
```
Comment (williamgoulois) [inline src/stories/pagination/Pagination.tsx]: Warning de eslint : 

`Unnecessarily computed property ['page-link-wrapper--active'] found. eslint(no-useless-computed-key)`

```typescript
 'page-link-wrapper--active': currentPage === page
```

File: src/stories/pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    useEffect(() => {
+        const paginationData: PaginationDataType = {
+            currentPage,
+            totalPages,
+            pageLimit,
+            totalRecords
+        }
+        onPageChanged !== undefined && onPageChanged(paginationData);
+    }, [currentPage]);
```
Comment (williamgoulois) [inline src/stories/pagination/Pagination.tsx]: Warning de eslint :

`React Hook useEffect has missing dependencies: 'onPageChanged', 'pageLimit', 'totalPages', and 'totalRecords'. Either include them or remove the dependency array. If 'onPageChanged' changes too often, find the parent component that defines it and wrap that definition in useCallback. eslint(react-hooks/exhaustive-deps)`

```typescript
}, [currentPage, onPageChanged, pageLimit, totalPages, totalRecords]);
```

**Review by williamgoulois — CHANGES_REQUESTED** [review]

File: src/stories/pagination/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+export type PaginationProps = {
+    totalRecords?: number;
+    pageLimit?: number;
+    onPageChanged?: (data: PaginationDataType) => void;
+}
+
+const LEFT_PAGE = 'LEFT';
+const RIGHT_PAGE = 'RIGHT';
```
Comment (1Green) [inline src/stories/pagination/Pagination.tsx]: Tu peux essayer d'utiliser un [enum](https://www.typescriptlang.org/docs/handbook/enums.html) en typescript ici !

File: src/stories/pagination/Pagination.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    totalRecords: { control: 'number' },
+  },
+} as Meta;
+
+const Template: Story<PaginationProps> = (args) => <Pagination {...args} />;
+
+export const Default = Template.bind({});
+Default.args = {
+    totalRecords: 1342
+}
```
Comment (1Green) [inline src/stories/pagination/Pagination.stories.tsx]: New line

**Review by 1Green — DISMISSED** [review]

**Comment by 1Green** [comment]
Deux petites remarques mais le reste semble ok

**Review by williamgoulois — DISMISSED** [review]

**Review by 1Green — DISMISSED** [review]

**Review by 1Green — APPROVED** [review]

**Review by williamgoulois — APPROVED** [review]

**Pull request merged**